ethereum/contracts/bridgehub/bridgehub-deps/BridgehubMailbox.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+// import {L2Log, L2Message} from "../chain-deps/ChainStorage.sol";
+import "./BridgehubBase.sol";
+import "../bridgehub-interfaces/IBridgehubMailbox.sol";
+import "../../state-transition/chain-interfaces/IStateTransitionChain.sol";
+
+contract BridgehubMailboxFacet is BridgehubBase, IBridgehubMailbox {
```

Comment:
Can BridgeHub use TransparentUpgreadable proxy pattern?

Reply:
It could in theory. However if we want the bridges to be backwards compatible, then the bridgehub will have to be the current Diamond, so it has to be a diamond proxy.

Reply:
@kelemeno I am not sure I follow. For bridges to be compatible we just have to provide the same interface, we don't need the Diamond. 

If I understand correctly, the "current" zkSync will become a StateTransitionChain (it can just transfer all the ETH to the SharedBridge). In this case, the DiamondProxy is indeed not really needed

Reply:
I was planning to have the current diamond become the Bridgehub, and then all external bridges will remain intact. 

This would break the external nodes, and apps that point to the proxy diamond. 

Alternatively we could do as you recommend, and have the diamond be a StateTransitionChain. In this case the bridges break, but the external nodes don't

Reply:
Discussed with Stas, now Bridgehub is a TUP